frontend/actions/index.js

```diff
@@ -0,0 +1,65 @@
+import { PipelineRequest, logger } from '@shopgate/engage/core';
+import { SMILE_DIGEST_PIPELINE, SWEETTOOTH_INIT_ENDPOINT } from '../constants';
+import {
+  requestSmileDigestData,
+  receiveSmileDigestData,
+  errorSmileDigestData,
+  clearSmileDigestData,
+} from '../action-creators';
+import getConfig from '../helpers/getConfig';
+
+const { smileJavascriptSdkSrc, smileChannelApiKey } = getConfig();
+/**
+ *
+ * @return {Function}
+ */
+export const fetchSmileDataDigest = () => (dispatch) => {
+  dispatch(requestSmileDigestData());
+  new PipelineRequest(SMILE_DIGEST_PIPELINE)
+    .dispatch()
+    .then(({ customerDigest, customerId }) => {
+      dispatch(receiveSmileDigestData(customerDigest, customerId));
+    })
+    .catch((error) => {
+      logger.error(error);
+      dispatch(errorSmileDigestData());
+    });
+};
+
+export const logoutSmile = () => (dispatch) => {
+  dispatch(clearSmileDigestData());
+  if (window.SmileUI) {
+    window.SmileUI.destroy();
+  }
+};
+
+/**
+ * Mount Smile script of reinitialize
+ * @param {string} externalCustomerId External customer id
+ * @param {string} digest Smile digest
+ * @return {Function}
+ */
+export const mountSmileScript = (externalCustomerId, digest) => async () => {
```

Comment:
maybe we should rename the function or split it because it seems to do more then just mounting the script

Reply:
renamed function